tests/fuzz/Cargo.toml

```diff
@@ -0,0 +1,28 @@
+[package]
+name = "typst-syntax-fuzz"
+version = "0.0.0"
+publish = false
+edition = "2021"
+
+[package.metadata]
+cargo-fuzz = true
+
+[dependencies]
+arbitrary = { version = "1.3.1", features = ["derive"] }
+libfuzzer-sys = "0.4"
+
+[dependencies.typst-syntax]
+path = "../../crates/typst-syntax"
+
+# Prevent this from interfering with workspaces
```

Comment:
can't this just be part of the workspace?

Reply:
Not really, this part is autogenerated by cargo-fuzz. I believe they are working on making this integrate nicer with existing workspace projects. https://github.com/rust-fuzz/cargo-fuzz/issues/345#issuecomment-1613634850,

Reply:
reading [this](https://github.com/rust-fuzz/cargo-fuzz/issues/345#issuecomment-1613495598) I'm not sure whether there's a real problem or if this is just them being cautious to have a working default setup where the main Cargo.toml doesn't need to be edited. have you tried removing the workspace here and adding the crate to the main workspace? deduplicating the Cargo.lock would be a big win imo.

Reply:
I haven't tried it with this specific project, but have run into a couple of issue previously. One of those issues being that you generally want to compile the main application/lib code against stable rustc. But because cargo-fuzz relies on having specific instrumentation flags available it's only possible to build fuzzer's on nightly. So if you include the fuzzing directory as part of the larger cargo workspace, then you'll run into issues unless you switch everything over to nightly. Would I be wrong in saying that you don't want to switch everything over to nightly?

I wonder if you can conditionally include crates in a workspace based on the rustc channel... That would largely solve the problem.

Reply:
You're right that I want to remain on stable. I wonder though, what kinds of issues arise when compiling the same workspace with stable and nightly?

Reply:
For one, I'm just not sure how to do that... I'm experimenting atm with [default-members](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-default-members-field) and the regular members field. I'm just not sure it's possible to compile on member on stable and another on nightly within the same build.

There's also some some issues where [cargo-fuzz adds specific instrumentation flags](https://github.com/rust-fuzz/cargo-fuzz/blob/a860fd92bc183f1fed45583aa78691b684e80576/src/project.rs#L157-L170). Which means that even if you can to a heterogenous build with stable/nightly there might be some weird side effects where things don't work as expected. I'm giving it a go now, but I'm unsure that you'd get "correct" fuzzing binaries, without building via cargo-fuzz.

Reply:
Maybe I'm missing something. But as long as the manifests are compatible with stable, can't the normal stable toolchain just compile everything but the fuzzer? And then, when fuzzing, the whole workspace is compiled with nightly?

Reply:
No I think your probably right, I'm only just revisiting cargo workspaces now after not really looking at the changes in docs for some time. So part of this could be me misremembering/being-plain-wrong. So thanks for asking the right questions to help me clear up my past assumptions/brain-baggage :)